tests/test_read_resample.py

```diff
@@ -98,6 +98,18 @@ def test_resampling_alg_error():
             src.read(1, out_shape=(1, 10, 10), resampling=Resampling.max)
 
 
+@requires_gdal31
+def test_resampling_sum():
+    """Test Resampling.sum method"""
+    with rasterio.open('tests/data/float.tif') as s:
+        out_shape = (2, 2)
+        weighted_sum = s.read(1, out_shape=out_shape, resampling=Resampling.sum)
+        expected = np.array([
+            [2.025, 1.395],
+            [0.205, -1.995]], dtype=np.float32)
+        assert (weighted_sum == expected).all()  # all True.
+
+
```

Comment:
This test can go. GDAL won't support this. Which is too bad, because it's a nice test.